packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -385,4 +385,21 @@ module.exports = ({ strapi }) => ({
 
     return sumDraftCounts(entity, uid);
   },
+
+  async getMultipleEntriesNumberOfDraftRelations(ids, uid) {
+    const { populate, hasRelations } = getDeepPopulateDraftCount(uid);
+
+    if (!hasRelations) {
+      return 0;
+    }
+
+    const totalNumberDraftRelations = await ids.reduce(async (acc, id) => {
+      const entity = await strapi.entityService.findOne(uid, id, { populate });
```

Comment:
I think you should findMany before the reduce and resolve in parallel with Promise.all then reduce on that result

Reply:
I didn't found a `strapi.entityService.findMany` method

Reply:
sorry found

Reply:
Do you mean something like this @markkaylor 
`    const entities = await strapi.entityService.findMany(uid, {
      populate,
    });

    const totalNumberDraftRelations = await entities.reduce(async (acc, entity) => {
      const sumDraftRelations = await sumDraftCounts(entity, uid);
      const prevCounter = await acc;
      return sumDraftRelations + prevCounter;
    }, 0);`

Reply:
I was thinking something like:

```js
    const entities = await strapi.entityService.findMany(uid, {
      populate,
      where: { id: { $in: ids } },
    });

    const totalNumberDraftRelations = entities.reduce((acc, entity) => {
      const entityDraftRelationCount = sumDraftCounts(entity, uid);
      const totalCount = acc + entityDraftRelationCount;

      return totalCount;
    }, 0);

```

Reply:
I added the where condition to the findMany method